### ISSUE TITLE
Device Support - WAXMAN LeakSMART v2

### ISSUE THREAD

**Issue by jtscott** [dresden-elektronik/deconz-rest-plugin#2701]
Hello I am opening this issue to request device support for the LeakSmart Smart Home Sensor 2.0. This is a water leak sensor and I hope to provide as much info as I can:

- [User Manual](https://leaksmart.com/storage/2019/10/LeakSmart-Smart-Home-Water-Sensor-User-Manual.pdf)
- [ZHA Device Handler](https://github.com/zigpy/zha-device-handlers/tree/c4fb37523e88f8ae743a9bc6c63f63090c539263/zhaquirks/waxman)

Device Screenshots
![waxman_leaksmart1](https://user-images.githubusercontent.com/154165/79620289-a64d7800-80cc-11ea-8278-70f5f1d9cb85.PNG)
![waxman_leaksmart2](https://user-images.githubusercontent.com/154165/79620290-a6e60e80-80cc-11ea-8005-9a503c513a80.PNG)
![waxman_leaksmart3](https://user-images.githubusercontent.com/154165/79620376-e9a7e680-80cc-11ea-810f-ffa94669ca32.PNG)
[debug.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/4495601/debug.txt)

Please let me know if I can provide any addition information, and thank you for your support and consideration.

**Comment by SwoopX** [comment]
Please also provide a screenshot of the node itself from deconz GUI where you can see the available clusters. Thanks!

**Comment by jtscott** [comment]
Sure, is this correct?
![image](https://user-images.githubusercontent.com/154165/79626525-06084b00-80ee-11ea-866f-d39e5741b37b.png)

**Comment by SwoopX** [comment]
It is, thanks. I'll let you know it I need anything else.

**Comment by jtscott** [comment]
Yes, I removed and re-added the node to be sure. When I had it paired to my wink hub a year ago it provided both leak detection and temperature.

**Comment by SwoopX** [comment]
Hm, then it moght be the unknown cluster. I'll check it out later.

**Comment by ebaauw** [comment]
Or a hidden _IAS Zone_ cluster.  Wouldn't be the first device to do that.

**Comment by SwoopX** [comment]
@jtscott Could you please also read the attributes from power configuration cluster and provide the screenshot? Wanna have the battery information right. Thanks!

**Comment by jtscott** [comment]
Sure thing; Here is what I am getting from it, it is updating the data from what I can see when I click read but not sure why the rest is grey. Device takes 3x AAA's physically:

![image](https://user-images.githubusercontent.com/154165/79989392-06497300-846d-11ea-9199-1946ae0eb5be.png)

**Comment by SwoopX** [comment]
Hm, can you read to try battery percentage once again (by double clicking on it) The 2 values seem not to fit each other.

**Comment by jtscott** [comment]
This looks weird for sure! I clicked read a couple times and it comes back 200 and says "reading done". I tried to read config as well and it comes back as not_found. Same with voltage, reading at 46 and read config not_found.
![image](https://user-images.githubusercontent.com/154165/79992043-37777280-8470-11ea-9f8a-8b98aae714bf.png)

**Comment by SwoopX** [comment]
It's just the "read" button. "read config" is to read the reporting configuration, which the device currently not has unless you did all the binding manually.

**Comment by jtscott** [comment]
No definitely outside my area of expertise there, I'm just capable of gui button clicks :) Reading still comes back 200: 
![image](https://user-images.githubusercontent.com/154165/79997252-9809ae00-8476-11ea-9d2c-0aa82384a611.png)

**Comment by SwoopX** [comment]
Hm, that still feels a bit strange 3x1.5V is not 46 (or 4.5) ;) Brand new batteries maybe?

Regardless, I'd rather go for the voltage for the time being. We can change that later on if required.

**Comment by jtscott** [comment]
Yes had to add new batteries last week, as this thing chirps loud like a fire alarm when its low.

**Comment by SwoopX** [comment]
Ok, we go for the percentage then as it doesn't require for any changes then. Just please keep an exe on if, if it changes over time.

**Comment by ebaauw** [comment]
> This looks weird for sure! 

No it doesn’t, check the ZCL spec:
- _Battery Percentage Remaining_ is in 0.5%, so a value of 200 means 100%.
- _Battery Voltage_ is in 0.1V, so a value of 46 means 4.6V.

If the reporting configuration returns _NOT FOUND_, it simply hasn’t been set.  Quite normal for factory new devices.  Try writing some values and reading them back.

**Comment by jtscott** [comment]
Hi Gents, I reset and re-added my sensor on the new release today. Looks like the leak part of the sensor is still missing. Is there something else I can provide you to help finish this device integration?

**Comment by SwoopX** [comment]
Just taken another looks and that one's gonna be interesting. You have 2 unknown clusters there, where the FC02 is a manufacturer specific cluster. You probably only have a chance to get some info out of it if you can compile an additional plugin and get some reasonable data back, which is not guaranteed.

Now, for 0b02, I recall I checked the specification already but haven't found anything. Just taken another look and this time, I was lucky as the zigbee specification contains a typo for that cluster. It's apparently "EN50523 Appliance Events and Alerts", so presumably responsible for the alerts. This is definitive a first timer here. Getting the cluster data in is the easy job, evaluating how it works in conjunction with the capabilities at hand is a different thing.

So long story short, we need to investigate if and how we can make it play.

**Comment by SwoopX** [comment]
Can any of you guys support in compiling the plugin and do some tests once all required parts have been implemented?

**Comment by jtscott** [comment]
Sure I can give it a shot, I'm probably savvy enough to follow specific instructions if you have those available?

**Comment by SwoopX** [comment]
You find the instructions in the readme of this repo. You just need to make sure you also installed the deconz-dev package. Hopefully, you don't run any virtualization (docker), otherwise it will be more difficult.

Give it a shot with the current state repo to see if it works out for you.

**Comment by jtscott** [comment]
switched over to dev today using the readme with no problem. I deleted the node from deconz and re-paired the sensor after factory reset but I don't see a difference. I looked over the clusters vs the old screenshots and it all looks the same.

Let me know what else I can do?

**Comment by SwoopX** [comment]
Hey, you were a bit too fast for me here ;) I'll prepare a seperate repo of mine for that particular testing so we stay kinda clean on the official one.

Take this one: https://github.com/SwoopX/st_playground.git
I already implemented code for water sensor creation. Currently working on the pick-up of the corresponsing alert commands...

**Comment by jtscott** [comment]
Woops sorry, haha! Ok gave it a shot this afternoon and looking really close! Attached is a debug where I removed the old sensor, re-added, and triggered two wet events on the sensor. The sensor makes a really annoying beep when it is wet so I am sure it fired.

[waxman_dbg.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/4713066/waxman_dbg.txt)

Device looks the same in deconz gui, but reflecting more info in webui:
![image](https://user-images.githubusercontent.com/154165/83446089-612a9e80-a40b-11ea-987d-026a9349a917.png)

Also showing up good in HASS, but of course no wet/dry events and sensor always reports dry right now:
![image](https://user-images.githubusercontent.com/154165/83446241-9c2cd200-a40b-11ea-8cc5-badc2e15a4da.png)

Hope that helps!

**Comment by SwoopX** [comment]
Alright, thanks for trying. No relevant output there, besides sensor creation.

Take 2 is out. Some more debug output and extended by one additional command. Please run it this time with `--dbg-info=2 --dbg-aps=1> dbg`, which hopefully gives me more insights in the command used.

**Comment by jtscott** [comment]
Sure thing! Here is a new debug output; I removed the device prior to updating and then re-added the device after compiling and triggered 2 water events again.

[dbg_waxman_20200602.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/4719850/dbg_waxman_20200602.txt)

**Comment by SwoopX** [comment]
Thanks. Sorry, but gotta ask: Have you pulled the repo once again and compiled? I don't see any debug output I'd expect...

**Comment by jtscott** [comment]
Hmm that's weird, yeah I removed prior cloned repos and re-cloned and re-compiled again. I can re-flash my SD back to before dev today and try again.

**Comment by SwoopX** [comment]
Just puzzled that I don't see anything from that particular cluster. Not sure anything else might be required upfront. However, I see at z2m that they seem to have added binding for the cluster in question.

Maybe give it a try when the sensor is paired:
**Manually bind a cluster**
In deconz GUI, select Panels and then Bind dropbox.
For your device as well as the coordinator (blue node), press the most right bullet to expand the available clusters.
For the coordinator: Drag&drop 01 Home Automation Endpoint as destination.
For the device: Select the windows covering cluster as source, also drag&drop it. Take note that this must be a server cluster (colored blue).
Press `bind`.

**Comment by jtscott** [comment]
ok did a couple things and it looks like the binding worked!

New debug after re-flashing to pre-dev, installing dev, and then recompiling from your repo here: 
[dbg_waxman_20200603.txt](https://github.com/dresden-elektronik/deconz-rest-plugin/files/4726094/dbg_waxman_20200603.txt)

Binding 0B02
![image](https://user-images.githubusercontent.com/154165/83685992-bbac3200-a5a6-11ea-954e-71d57dec30b0.png)

Binding FC02
![image](https://user-images.githubusercontent.com/154165/83686060-d088c580-a5a6-11ea-8542-6fe89f84b554.png)

Now... after the fact I thought I should have just did one cluster binding first to see what exactly happens but let me know if that's needed and I can repeat the process above :)

Thanks!

**Comment by SwoopX** [comment]
Better now 👍 Will check it later in detail.

**Comment by SwoopX** [comment]
Looks like we got a winner here:

```
03:31:07:839 APS-DATA.indication srcAddr: 0xfc14, srcEp: 0x01 dstAddrMode: 2, profile: 0x0104, cluster: 0x0B02, lqi: 255, rssi: -46
03:31:07:840 [APPL] Command is cluster command
03:31:07:840 [APPL] Alert count: 1
03:31:07:840 [APPL] Alert Structure: 4481
03:31:07:840 [APPL] Device currently alerting...
03:31:07:840 [APPL] Settings sensor state alert to 'true'...
03:31:07:842 Websocket 192.168.1.15:58846 send message: {"e":"changed","id":"12","r":"sensors","state":{"lastupdated":"2020-06-03T20:31:07.840","water":true},"t":"event","uniqueid":"00:22:a3:00:00:0f:e6:4a-01-0b02"} (ret = 159)
```

```
03:31:17:832 APS-DATA.indication srcAddr: 0xfc14, srcEp: 0x01 dstAddrMode: 2, profile: 0x0104, cluster: 0x0B02, lqi: 255, rssi: -46
03:31:17:832 [APPL] Command is cluster command
03:31:17:832 [APPL] Alert count: 1
03:31:17:832 [APPL] Alert Structure: 385
03:31:17:832 [APPL] Device ceased alert...
03:31:17:832 [APPL] Settings sensor state alert to 'false'...
03:31:17:834 Websocket 192.168.1.15:58846 send message: {"e":"changed","id":"12","r":"sensors","state":{"lastupdated":"2020-06-03T20:31:17.832","water":false},"t":"event","uniqueid":"00:22:a3:00:00:0f:e6:4a-01-0b02"} (ret = 160)
```

**Comment by jtscott** [comment]
Yep looks like 0x0b02. I unbound the cluster and made the sensor wet and it did not trigger a state change. Rebinding 0x0b02 worked again and it triggers a wet state change.

**Comment by SwoopX** [comment]
Alright. Updated the code so the binding should take place automatically. If you could test once more and it succeeds, I guess we're done here.

**Comment by jtscott** [comment]
Ok all done and can confirm works good. I reflashed to pre-dev, recompiled your repo, then re-added the sensor. Works out of box now!

Thanks so much, really appreciated!

**Comment by arjewell23** [comment]
Hi Guys,  I've been following this because I have a few leaksmart sensors.  How can I get this to work?  Do I need to upload some files, etc?

**Comment by arjewell23** [comment]
I have my HA instance installed on a VM,  Will this still work?

**Comment by SwoopX** [comment]
It should, if you don't use docker.

Added the code to the PR. Enjoy!

**Comment by SwoopX** [comment]
Closing this as PR has been merged. Supported with .78

**Comment by wkearney99** [comment]
Anyone had one of these sensors trip a false detection of moisture?  I've had one running for about a year.  The battery level was down to about 10%.  It detected water (and there is none).  I changed out the batteries, and the gateway now sees the new AAAs as 100%, but it's still reading 'Wet'.

Is there anything else I have to do to clear the error?

**Comment by wkearney99** [comment]
And nevermind, I wet the unit and it alarmed and when I dried it the condition went back to Dry.

So apparently batteries alone will not clear it?
